### DISCUSSION TITLE
Is it possible to use vercel as a cdn only?

### DISCUSSION THREAD

**Discussion by arso1er** [vercel/next.js#20248] [Help] [answered]
Hi,

I'm working on a nextjs app with a custom server using  cloudflare as cdn. When I run tests with google insight, it tells me to reduce  my server response time. That seemed weird to me because I'm using getStaticProps. After searching, I found that cloudflare doesn't prioritize time to first byte, which could be the culprit. I made a quick deployment to vercel and my app score gained more than 10 points on mobile.

So I'd like to know if there's a way to use vercel as a cdn only (like cloudflare) while hosting my app on my own server.

Thanks.

**Comment by leerob** [comment, marked as answer]
Yes, you can use Vercel as a CDN while deploying your Next.js app. It works out of the box using the _severless_ paradigm.

Using a custom server (e.g. Node server) is breaking from the serverless model because it requires an always-on server. Vercel only hosts serverless applications.

You can host your backend elsewhere and still use Vercel as a CDN for your front-end Next.js application. Especially if you're using `getStaticProps`, then you aren't actually talking to the server at request time at all. The file being served to your users is static HTML cached at Vercel's edge.

Hope this helps.

**Reply by arso1er** [reply]
So if I understand correctly, I'll have to separate my app in two parts (an express api and a nextjs frontend) if I want to use Vercel as CDN. I'll stay with cloudflare for now then and maybe think more about it later.

Thanks for your answer.

**Reply by leerob** [reply]
I don't believe Cloudflare can host Node.js backends, but I could be wrong.

**Reply by arso1er** [reply]
I don't think they can either. Currently, my app is a nextjs app with a custom express server. I deployed my app on a digitalocean vps. Then I routed my web traffic through the Cloudflare network, by updating the nameservers at my domain registrar to resolve my domain’s DNS with Cloudflare's nameservers.

I was wondering if I could do something similar with Vercel CDN instead of cloudflare, but from your answer I think it's not possible unless I split my app in two parts.

**Comment by MasteringBOX** [comment]
Hi @arso1er  and @leerob, 

I have a follow up question regarding this topic. In Nextjs you can configure a CDN with assetPrefix. Can I configure Vercel to be my CDN while deploying the nextjs app in my server?. 

I need to have nginx proxy for a set of rules to redirect/rewrite to other microservices in the same domain, so I can't really deploy in Vercel in the same domain as my other apps -- or at least I don't know how to. 

Any idea how I could leverage Vercel CDN for my app? 

Regards
